bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+import org.openhab.binding.salus.internal.rest.GsonMapper;
+import org.openhab.binding.salus.internal.rest.JettyHttpClient;
+import org.openhab.binding.salus.internal.rest.SalusApi;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public final class CloudBridgeHandler extends BaseBridgeHandler implements CloudApi {
+    private Logger logger = LoggerFactory.getLogger(CloudBridgeHandler.class.getName());
+    private final HttpClientFactory httpClientFactory;
+    @NonNullByDefault({})
+    private LoadingCache<String, SortedSet<DeviceProperty<?>>> devicePropertiesCache;
+    @Nullable
+    private SalusApi salusApi;
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            internalInitialize();
+            updateStatus(ONLINE);
+        } catch (Exception ex) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "Cannot start server! " + ex.getMessage());
+        }
+    }
+
+    private void internalInitialize() {
+        var config = this.getConfigAs(CloudBridgeConfig.class);
+        var missingUsername = !config.hasUsername();
+        var missingPassword = !config.hasPassword();
+        if (missingUsername || missingPassword) {
+            var sb = new StringBuilder();
+            sb.append("Missing configuration!\n");
+            sb.append(missingUsername ? "❌" : "✅").append(" username\n");
+            sb.append(missingPassword ? "❌" : "✅").append(" password\n");
+            sb.append("Please check your configuration!\n");
+            logger.error("{}", sb);
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, sb.toString());
+            return;
+        }
+        var httpClient = new JettyHttpClient(httpClientFactory.getCommonHttpClient());
+        var localSalusApi = salusApi = new SalusApi(config.getUsername(), config.getPassword().toCharArray(),
+                config.getUrl(), httpClient, GsonMapper.INSTANCE);
+        logger = LoggerFactory
+                .getLogger(CloudBridgeHandler.class.getName() + "[" + config.getUsername().replace(".", "_") + "]");
+        try {
+            localSalusApi.findDevices();
+        } catch (Exception ex) {
+            var msg = "Cannot connect to Salus Cloud! Probably username/password mismatch!";
+            logger.error(msg, ex);
```

Comment:
See earlier ccomment in previous round, please check all occurrences.
Not needed as updateStatus is allready called. Please check all logging if it is the right level and acording to the reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
done